pkg/artifactory/resource_artifactory_push_replication.go

```diff
@@ -95,6 +102,10 @@ var pushReplicationSchema = map[string]*schema.Schema{
 		Type:     schema.TypeString,
 		Optional: true,
 	},
+	"proxy": {
+		Type:     schema.TypeString,
+		Optional: true,
```

Comment:
Description. Also, if I am not mistaken, if a blank string is sent it doesn't actually change anything @alexhung

Reply:
Correct. If empty string (or `null` in JSON payload) is sent to Artifactory, it will be ignored. To reset `proxy` to nothing, you need to send an non-existent proxy key. Anything will do as long as it isn't empty string or `null`.

You can use this function to do what you need: https://github.com/jfrog/terraform-provider-artifactory/blob/master/pkg/artifactory/repositories.go#L847